sdcm/mgmt.py

```diff
@@ -4,10 +4,292 @@
+        assert self.status != TaskStatus.RUNNING
+
+    def start(self):
+        cmd = "task start --continue {} -c {}".format(self.id, self.mgr_cluster.id)
+        res = self.mgr_tool.run_sctool_cmd(cmd=cmd, is_verify_errorless_result=True)
+        assert self.status not in [TaskStatus.STOPPED]
+
+    @property
+    def status(self):
+        """
+        Gets the repair task's status
+        """
+        cmd = "task list -c {}".format(self.mgr_cluster.id)
+        res = self.mgr_tool.run_sctool_cmd(cmd=cmd, is_verify_errorless_result=True)
+        if self.id not in res.stdout:
+            logger.error("Encountered an error on '{}' command response: {}".format(cmd, str(res.stdout)))
+            raise ScyllaManagerError("Encountered an error on '{}' command response".format(cmd))
+        lines = res.stdout.split('\n')
+        status = TaskStatus.UNKNOWN
+        for line in lines:
+            if self.id in line:
+                raw_status = line.split()[-2]
+                status = TaskStatus.from_str(raw_status)
+                break
+        logger.debug("Task: {} status is: {}".format(self.id,str(status)))
+        return status
+
+    @property
+    def progress(self):
+        """
+        Gets the repair task's progress
+        """
+        if self.status == TaskStatus.NEW:
+            return "0%"
+        cmd = "task progress {} -c {}".format(self.id, self.mgr_cluster.id)
+        res = self.mgr_tool.run_sctool_cmd(cmd=cmd)
+        if not res or "Progress" not in res.stdout:
+            logger.error("Encountered an error on '{}' command response: {}".format(cmd, str(res)))
+            raise ScyllaManagerError("Encountered an error on '{}' command response".format(cmd))
+        lines = res.stdout.split('\n')
+        progress = 'N/A'
+        for line in lines:
+            if "Progress" in line:
+                progress = line.split()[3]
+                break
+        logger.debug("Task: {} progress is: {}".format(self.id,progress))
+        return progress
+
+    def is_status_in_list(self, list_status, check_task_progress=False):
+        """
+        Check if the status of a given task is in list
+        :param list_status:
+        :return:
+        """
+        status = self.status
+        if check_task_progress and status != TaskStatus.NEW: # check progress for all statuses except 'NEW'
+            progress = self.progress
+        return self.status in list_status
+
+    def wait_for_status(self, list_status, check_task_progress=True):
+        text = "Waiting until task: {} reaches status of: {}".format(self.id, list_status)
+        is_status_reached = False
+        is_status_reached = wait.wait_for(func=self.is_status_in_list, step=60,
+                                          text=text, list_status=list_status, check_task_progress=check_task_progress, timeout=3600)
+        return is_status_reached
+
+    def wait_for_status_done(self):
```

Comment:
It should have a timeout parameter.

Reply:
added: def wait_for_status_done(self, timeout=3600):